### PR TITLE
adding title bar support for mac

### PR DESCRIPTION
**Pull request by devaradhanm**
#### Description of changes

- As part of this PR, we handle mac os separately. If mac, we use window frame with titlebar style set to hidden. This adds 3 icons to top left corner of window. So, we need to add some padding to the title bar to cover this icons. Also, we remove the title bar if the window is in full screen mode.

- For windows & mac, we show the minimize, close & maximize icons on the right side.

- As part of this change, we have also made changes to only listen to maximize, unmaximize, enter-full-screen & leave-full-screen. We don't have to listen to minimize action, since we don't change state based on whether the window is minimized or not.

**Device connect view:**
image

**Results view in maximized mode:**
image

**Results view in full screen view**
image

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/electron/main/main.ts [hunk 1]
```diff
@@ -27,11 +32,11 @@ const createWindow = () => {
 };
 
 const enableDevMode = (window: BrowserWindow) => {
-    if (process.env.DEV_MODE === 'true') {
-        window.webContents.openDevTools({
-            mode: 'detach',
-        });
-    }
+    // if (process.env.DEV_MODE === 'true') {
```
Comment (Shobhit1) [inline src/electron/main/main.ts]: is this intentional?

File: src/electron/views/device-connect-view/components/window-title.tsx [hunk 2]
```diff
@@ -1,32 +1,48 @@
...
+    }
+
+    const windowtitleClassNames = [windowTitle, props.className].filter(c => c != null);
+
+    if (props.deps.platformInfo.isMac()) {
+        windowtitleClassNames.push(macWindowTitle);
+    }
     return (
-        <header className={[windowTitle, props.className].filter(c => c != null).join(' ')}>
+        <header className={windowtitleClassNames.join(' ')}>
```
Comment (smoralesd) [inline src/electron/views/device-connect-view/components/window-title.tsx]: so... this line: 
```typescript
 const windowtitleClassNames = [windowTitle, props.className].filter(c => c != null);
```
and this line 
```typescript
windowtitleClassNames.join(' ')
```
is it worth it to use office fabric `css` function? (which does the same)

File: src/tests/electron/window-frame-listener.test.ts [hunk 3]
```diff
@@ -20,75 +20,67 @@ describe(WindowFrameListener, () => {
...
-            eventsToListen.forEach(eventName => {
-                browserWindowMock
-                    .setup(b => b.on(eventName as any, It.isAny()))
-                    .callback((event, cb) => {
-                        eventCallbacks.push(cb);
-                    })
-                    .verifiable(Times.once());
-            });
+        beforeEach(() => {
+            setupVerifiableWindowEventCallback('maximize', cb => (maximizeCallback = cb));
```
Comment (smoralesd) [inline src/tests/electron/window-frame-listener.test.ts]: should each test set mocks only for what each test need? 
Having all mocks set up for all cases make the code a little hard to follow as in, I have no idea which mock is really needed for each unit test.

**Comment by Shobhit1** [comment]
I feel like this PR can be broken into at least 2 more small PRs accomplishing the subtasks that you listed in the PR description. Is that too much work if we want to separate out this PR into smaller PRs? Please check.

File: src/tests/electron/window-frame-listener.test.ts (hunk 3 above)
Comment (devaradhanm) [reply src/tests/electron/window-frame-listener.test.ts]: i made the browserWindowMock to be strict to control the access to browser window. So, I may need to do the mock on all the types anyway. Or i need to have 1 test with strict mock on initialize & then loose mock for others. Do you prefer that?

**Comment by devaradhanm** [comment]
@Shobhit1 , I agree. I could split this in to two prs. I open to make this change. I could have split platform-info & window listener changes in to 1 pr & then remaining in another pr. I ended up converting my spike to productizable code.
